examples/AspNetCore/TelemetryService.cs

```diff
@@ -0,0 +1,38 @@
+// <copyright file="TelemetryService.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace Examples.AspNetCore;
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+public class TelemetryService : ITelemetryService
+{
+    private const string InstrumentationScopeName = "Examples.AspNetCore";
+
+    public TelemetryService(string version)
```

Comment:
I'd recommend not having a version parameter here. Its not clear to me we are gaining any value by parameterizing it. We can inline `typeof(TelemetryService).Assembly.GetName().Version?.ToString()` below.

Reply:
I parameterized it so that it was guaranteed to match the version defined as part of the provider's resource. Not that I would necessarily expect either of these to change. 

Slightly related but I dislike how the original code was setting "unknown" in the case of null since it is an optional parameter and already handled nicely.